addons/hw_posbox_homepage/static/src/app/components/dialog/WifiDialog.js

```diff
@@ -51,25 +62,40 @@ export class WifiDialog extends Component {
         }
 
         this.state.waitRestart = true;
-        const data = await this.store.rpc({
+        const responsePromise = this.store.rpc({
             url: "/hw_posbox_homepage/update_wifi",
             method: "POST",
             params: this.form,
         });
-
-        if (data.status !== "success") {
+        if (this.isCurrentlyConnectedToEthernet()) {
+            const data = await responsePromise;
+            if (data.status !== "success") {
+                this.state.waitRestart = false;
+            }
+        } else {
+            // The IoT box is no longer reachable, so we can't await the response.
+            this.state.statusMessage = `IoT Box is now connected to ${this.form.essid}. You may close this page.`;
```

Comment:
the message is being displayed even if a wrong password is entered and the IoT Box isn't connected to the Wifi